src/unity/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -36,95 +36,32 @@ static std::map<std::string,size_t> generate_column_index_map(const std::vector<
  *  Find the statistical mode (majority value) of a given vector.
  *
  * \param[in] input_vec a vector for which the mode will be calculated
-
  * \return    The most frequent value within the given vector.
  */
 
-static double vec_mode(const flex_vec& input_vec) {
-    std::vector<int> histogram;
-    for (size_t i = 0; i < input_vec.size(); ++i) {
-        size_t value = static_cast<size_t>(input_vec[i]);
-
-        // Each value should be in the index of a class label
-        DASSERT_EQ(static_cast<double>(static_cast<size_t>(input_vec[i])), input_vec[i]);
-
-        if(histogram.size() < (value + 1)){
-          histogram.resize(value + 1);
-        }
-
-        histogram[value]++;
-    }
+static double vec_mode(flex_vec::const_iterator first,
+                       flex_vec::const_iterator last) {
+  std::vector<int> histogram;
```

Comment:
Looks like you're putting in `size_t` in the vector. It would probably be best if it was an `std::vector` of `size_t`.